cpp/src/detail/shuffle_vertices.cu

```diff
@@ -249,4 +249,74 @@ shuffle_ext_vertex_value_pairs_to_local_gpu_by_vertex_partitioning(
+                                     rmm::device_uvector<int32_t>&& vertices,
+                                     rmm::device_uvector<size_t>&& values);
+
+template std::tuple<rmm::device_uvector<int32_t>, rmm::device_uvector<float>>
+shuffle_external_vertices_and_values(raft::handle_t const& handle,
+                                     rmm::device_uvector<int32_t>&& vertices,
+                                     rmm::device_uvector<float>&& values);
+
+template std::tuple<rmm::device_uvector<int32_t>, rmm::device_uvector<double>>
+shuffle_external_vertices_and_values(raft::handle_t const& handle,
+                                     rmm::device_uvector<int32_t>&& vertices,
+                                     rmm::device_uvector<double>&& values);
+
+template std::tuple<rmm::device_uvector<int64_t>, rmm::device_uvector<int32_t>>
+shuffle_external_vertices_and_values(raft::handle_t const& handle,
+                                     rmm::device_uvector<int64_t>&& vertices,
+                                     rmm::device_uvector<int32_t>&& values);
+
+template std::tuple<rmm::device_uvector<int64_t>, rmm::device_uvector<int64_t>>
+shuffle_external_vertices_and_values(raft::handle_t const& handle,
+                                     rmm::device_uvector<int64_t>&& vertices,
+                                     rmm::device_uvector<int64_t>&& values);
+
+template std::tuple<rmm::device_uvector<int64_t>, rmm::device_uvector<size_t>>
+shuffle_external_vertices_and_values(raft::handle_t const& handle,
+                                     rmm::device_uvector<int64_t>&& vertices,
+                                     rmm::device_uvector<size_t>&& values);
+
+template std::tuple<rmm::device_uvector<int64_t>, rmm::device_uvector<float>>
+shuffle_external_vertices_and_values(raft::handle_t const& handle,
+                                     rmm::device_uvector<int64_t>&& vertices,
+                                     rmm::device_uvector<float>&& values);
+
+template std::tuple<rmm::device_uvector<int64_t>, rmm::device_uvector<double>>
+shuffle_external_vertices_and_values(raft::handle_t const& handle,
+                                     rmm::device_uvector<int64_t>&& vertices,
+                                     rmm::device_uvector<double>&& values);
+
+template <typename vertex_t>
+rmm::device_uvector<vertex_t> shuffle_external_vertices(raft::handle_t const& handle,
+                                                        rmm::device_uvector<vertex_t>&& vertices)
+{
+  return detail::shuffle_ext_vertices_to_local_gpu_by_vertex_partitioning(handle,
+                                                                          std::move(vertices));
+}
+
+template rmm::device_uvector<int32_t> shuffle_external_vertices(
+  raft::handle_t const& handle, rmm::device_uvector<int32_t>&& d_vertices);
+
+template rmm::device_uvector<int64_t> shuffle_external_vertices(
+  raft::handle_t const& handle, rmm::device_uvector<int64_t>&& d_vertices);
```

Comment:
It will be more natural to list `shuffle_external_vertices` before `shuffle_external_vertices_and_values`

Reply:
Changed.